cilium/bpf_metadata.cc

```diff
@@ -406,38 +406,42 @@ Cilium::SocketOptionSharedPtr Config::getMetadata(Network::ConnectionSocket& soc
     // North/south L7 LB, assume the source security identity of the configured source addresses,
     // if any and policy for this identity exists.
 
-    // Pick the local source address of the same family as the incoming connection
-    const Network::Address::Ip* ip = selectIPVersion(sip->version(), source_addresses);
-
-    if (!ip) {
-      // IP family of the connection has no configured local source address
-      ENVOY_LOG(warn,
-                "cilium.bpf_metadata (north/south L7 LB): No local IP source address configured "
-                "for the family of {}",
-                pod_ip);
+    // Pick the local ingress source address of the same family as the incoming connection
+    const Network::Address::Ip* ingress_ip = selectIPVersion(sip->version(), source_addresses);
+    auto& ingress_ip_str = ingress_ip->addressAsString();
```

Comment:
This would crash if `ingress_ip` is `nullptr`, tested above. So have to move after that test.

Reply:
glad that i have your second pair of eyes this morning :facepalm: thanks